### PR TITLE
Remove`capture_output`, rely on pytests.

### PR DESCRIPTION
**Pull request by stephenroller**
**Patch description**
The switch to pytests gives us output capturing and debug friendly for free. On a failed test, stdout is automatically presented to the user. The user can also always force it on with the `-s` flag: `python -m pytest -s`.

**Testing**
CI should pass.

### CODE REVIEW

File: tests/nightly/gpu/test_bert.py [hunk 1]
```diff
@@ -63,12 +63,12 @@ def test_crossencoder(self):
         self.assertGreaterEqual(
             test['accuracy'],
             0.03,
-            'test accuracy = {}\nLOG:\n{}'.format(test['accuracy'], stdout),
+            'test accuracy = {}\nLOG:\n{}'.format(test['accuracy']),
```
Comment (stephenroller) [inline tests/nightly/gpu/test_bert.py]: leak

File: tests/nightly/gpu/test_bert.py [hunk 2]
```diff
@@ -37,12 +37,12 @@ def test_biencoder(self):
         self.assertLessEqual(
             test['accuracy'],
             0.5,
-            'test accuracy = {}\nLOG:\n{}'.format(test['accuracy'], stdout),
+            'test accuracy = {}\nLOG:\n{}'.format(test['accuracy']),
```
Comment (stephenroller) [inline tests/nightly/gpu/test_bert.py]: leak

File: tests/nightly/gpu/test_bert.py [hunk 3]
```diff
@@ -63,12 +63,12 @@ def test_crossencoder(self):
...
             test['accuracy'],
             0.03,
-            'test accuracy = {}\nLOG:\n{}'.format(test['accuracy'], stdout),
+            'test accuracy = {}\nLOG:\n{}'.format(test['accuracy']),
         )
         self.assertLessEqual(
             test['accuracy'],
             0.8,
-            'test accuracy = {}\nLOG:\n{}'.format(test['accuracy'], stdout),
+            'test accuracy = {}\nLOG:\n{}'.format(test['accuracy']),
```
Comment (stephenroller) [inline tests/nightly/gpu/test_bert.py]: leak

File: tests/nightly/gpu/test_drqa.py [hunk 4]
```diff
@@ -15,16 +15,16 @@ class TestDrQAModel(unittest.TestCase):
...
     def test_pretrained(self):
-        stdout, _, test = testing_utils.eval_model(
+        _, test = testing_utils.eval_model(
             dict(task='squad:index', model_file='zoo:drqa/squad/model')
         )
         self.assertGreaterEqual(
             test['accuracy'],
             0.68,
-            'test accuracy = {}\nLOG:\n{}'.format(test['accuracy'], stdout),
+            'test accuracy = {}\nLOG:\n{}'.format(test['accuracy']),
```
Comment (stephenroller) [inline tests/nightly/gpu/test_drqa.py]: leak

File: tests/nightly/gpu/test_drqa.py [hunk 5]
```diff
@@ -15,16 +15,16 @@ class TestDrQAModel(unittest.TestCase):
...
         )
         self.assertGreaterEqual(
-            test['f1'], 0.78, 'test f1 = {}\nLOG:\n{}'.format(test['f1'], stdout)
+            test['f1'], 0.78, 'test f1 = {}\nLOG:\n{}'.format(test['f1'])
```
Comment (stephenroller) [inline tests/nightly/gpu/test_drqa.py]: leak

File: tests/nightly/gpu/test_transresnet.py [hunk 6]
```diff
@@ -32,37 +32,28 @@ def setUpClass(cls):
...
     def test_transresnet(self):
         """
         Test pretrained model.
         """
         stdout, _, test = testing_utils.eval_model(MODEL_OPTIONS, skip_valid=True)
         self.assertEqual(
-            test['accuracy'],
-            0.4,
-            'test accuracy = {}\nLOG:\n{}'.format(test['accuracy'], stdout),
+            test['accuracy'], 0.4, 'test accuracy = {}'.format(test['accuracy']),
```
Comment (stephenroller) [inline tests/nightly/gpu/test_transresnet.py]: can be even simpler

File: tests/tasks/test_wizard_of_wikipedia.py [hunk 7]
```diff
@@ -67,24 +67,23 @@ def test_output(self):
...
 
-    def run_display_test(self, kwargs):
-        f = io.StringIO()
-        with redirect_stdout(f):
+    def _run_display_test(self, kwargs):
+        with testing_utils.capture_output() as stdout:
             parser = setup_args()
             parser.set_defaults(**kwargs)
-            opt = parser.parse_args()
+            opt = parser.parse_args([])
```
Comment (stephenroller) [inline tests/tasks/test_wizard_of_wikipedia.py]: this should be in the previous PR.

File: parlai/core/metrics.py [hunk 8]
```diff
@@ -361,7 +361,7 @@ def update(self, observation, labels):
                 for i in range(3):
                     weights = [1 / (i + 1) for _ in range(i + 1)]
                     bleu_scores[f'bleu-{i + 1}'] = _bleu(prediction, labels, weights)
-            if 'rouge-L' in self.metrics_list:
+            if 'rouge-L' in self._print_metrics_list:
```
Comment (emilydinan) [inline parlai/core/metrics.py]: ... where did this come from?

File: tests/test_build_data.py [hunk 9]
```diff
@@ -64,19 +58,17 @@ def test_download_multiprocess_chunks(self):
...
+        )
 
         output_filenames, output_statuses, output_errors = zip(*download_results)
 
-        self.assertIn('mnist0.tar.gz', output_filenames, f'missing file:\n{stdout}')
-        self.assertIn('mnist1.tar.gz', output_filenames, f'missing file:\n{stdout}')
-        self.assertIn('mnist2.tar.gz', output_filenames, f'missing file:\n{stdout}')
-        self.assertIn(200, output_statuses, f'unexpected error code:\n{stdout}')
-        self.assertIn(403, output_statuses, f'unexpected error code:\n{stdout}')
+        self.assertIn('mnist0.tar.gz', output_filenames, 'missing file mnist0.tar.gz')
```
Comment (emilydinan) [inline tests/test_build_data.py]: do you need these error messages here?

Comment (emilydinan) [reply tests/test_build_data.py]: why not just get rid of them

File: parlai/core/metrics.py (hunk 8 above)
Comment (stephenroller) [reply parlai/core/metrics.py]: From noticing the test wasn’t working as intended

**Pull request merged**